system/security/digisign/digisign.cpp

```diff
@@ -0,0 +1,224 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2012 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+#include "jliball.hpp"
+#include "build-config.h"
+#include <openssl/pem.h>
+#include <openssl/err.h>
+#include "digisign.hpp"
+
+static CriticalSection digiSignCrit;
+static CriticalSection digiVerifyCrit;
+
+
+
+class _CDigitalSignatureManager : implements IDigitalSignatureManager
+{
+private:
+    StringAttr   publicKeyFile;
+    StringAttr   privateKeyFile;
+    bool         signingConfigured;
+    bool         verifyingConfigured;
+
+public:
+    _CDigitalSignatureManager()
+    {
+        signingConfigured = false;
+        verifyingConfigured = false;
+#ifdef _USE_ZLIB
+        //query private key file location from environment.conf
+        StringBuffer configFileSpec;
+        configFileSpec.set(CONFIG_DIR).append(PATHSEPSTR).append("environment.conf");
+        Owned<IProperties> conf = createProperties(configFileSpec.str(), true);
+        if (conf)
+        {
+            publicKeyFile.set(conf->queryProp("HPCCPublicKey"));
+            privateKeyFile.set(conf->queryProp("HPCCPrivateKey"));
+            signingConfigured = !publicKeyFile.isEmpty();
+            verifyingConfigured = !privateKeyFile.isEmpty();
+        }
```

Comment:
Same code is in pkencrypt lib - can we common up somehow in environment.cpp or somewhere ?
Or at least use defines for "HPCCPublicKey" and "HPCCPrivateKey" in one file.